eva-server/src/main/java/uk/ac/ebi/eva/server/ws/StudyWSServer.java

```diff
@@ -67,21 +71,21 @@ public QueryResponse getFilesByStudy(@PathVariable("study") String study,
             throws UnknownHostException, IllegalOpenCGACredentialsException, IOException {
         initializeQueryOptions();
 
-        StudyDBAdaptor studyMongoDbAdaptor = DBAdaptorConnector.getStudyDBAdaptor(species);
-        VariantSourceDBAdaptor variantSourceDbAdaptor = DBAdaptorConnector.getVariantSourceDBAdaptor(species);
+        MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName(species));
+
+        QueryResult queryResult = Utils.buildQueryResult(variantSourceEntityRepository.findByStudyId(study));
 
-        QueryResult idQueryResult = studyMongoDbAdaptor.findStudyNameOrStudyId(study, queryOptions);
-        if (idQueryResult.getNumResults() == 0) {
-            QueryResult queryResult = new QueryResult();
+        if (queryResult.getResult().size() == 0) {
+            queryResult = Utils.buildQueryResult(Collections.emptyList());
             queryResult.setErrorMsg("Study identifier not found");
             response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
             return setQueryResponse(queryResult);
         }
 
-        BasicDBObject id = (BasicDBObject) idQueryResult.getResult().get(0);
-        QueryResult finalResult = variantSourceDbAdaptor.getAllSourcesByStudyId(
-                id.getString(DBObjectToVariantSourceConverter.STUDYID_FIELD), queryOptions);
-        finalResult.setDbTime(finalResult.getDbTime() + idQueryResult.getDbTime());
+        VariantSourceEntity variantSourceEntity = (VariantSourceEntity) queryResult.getResult().get(0);
+        QueryResult finalResult
+                = Utils.buildQueryResult(variantSourceEntityRepository.findByStudyId(variantSourceEntity.getStudyId()));
```

Comment:
In fact, I think we could still just do only the first query `findByStudyIdOrStudyName`